packages/ui/src/components/Table/BaseTable.test.tsx

```diff
@@ -660,4 +660,95 @@ describe('<Table />', () => {
 
         expect(mockOnRowClick).toHaveBeenCalledTimes(1);
     });
+
+    it('renders sticky styles when cell.isSticky is true and stickyPosition is defined', () => {
+        const { getByTestId } = render(<Table headCells={headCellsWithSticky} rows={rowsWithSticky.slice(0, 1)} />);
+        const stickyCellId = Object.keys(headCellsWithSticky).find(
+            (cellId) =>
+                headCellsWithSticky[cellId as keyof typeof headCellsWithSticky].isSticky &&
+                headCellsWithSticky[cellId as keyof typeof headCellsWithSticky].stickyPosition
+        );
+
+        if (stickyCellId) {
+            const renderedCell = getByTestId(`table-row-cell-${stickyCellId}`);
+            const { stickyPosition } = headCellsWithSticky[stickyCellId as keyof typeof headCellsWithSticky];
+            if (stickyPosition)
+                expect(renderedCell).toHaveStyle({
+                    position: 'sticky',
+                    [stickyPosition]: 0,
+                });
+            else throw new Error('No stickyPosition found in headCellsWithSticky');
+        } else {
+            throw new Error('No sticky cells found in headCellsWithSticky');
+        }
+    });
+
+    it('applies sticky styles to the head cell when isSticky is true and stickyPosition is defined', () => {
+        const { getByTestId } = render(<Table headCells={headCellsWithSticky} rows={rowsWithSticky.slice(0, 1)} />);
```

Comment:
since that is your mocked test data, you always know which columns are sticky, no need to do a `find`. If you want, you can even declare the mock data in the `it` block so that is closer to the test.

In this way you eliminate the if's and throw's